pypesto/objective/priors.py

```diff
@@ -225,6 +228,51 @@ def residual_jacobian(self, x):
 
         return sres
 
+    def sample(
+        self, n_samples: int = 1, seed: int = None
+    ) -> dict[str, np.ndarray]:
+        """
+        Sample from the prior distribution.
+
+        Parameters
+        ----------
+        n_samples:
+            Number of samples to draw.
+        seed:
+            Random seed for reproducibility.
+
+        Returns
+        -------
+        samples:
+            Dictionary of samples from the prior distributions for each
+            parameter index.
+        """
+        rng = np.random.default_rng(seed)
+
+        # Find the maximum index in prior_list
+        prior_samples_dict = {
+            prior["index"]: np.zeros(n_samples) for prior in self.prior_list
+        }
+        # n_parameters = max(prior["index"] for prior in self.prior_list) + 1
+        # samples = np.zeros((n_samples, n_parameters)) * np.nan
+
+        for prior in self.prior_list:
+            index = prior["index"]
+            prior_type = prior.get("type", None)
+            prior_parameters = prior.get("parameters", None)
+
+            if prior_type is None or prior_parameters is None:
+                raise ValueError(
+                    "Prior type and parameters must be specified for sampling. "
+                    "Use 'get_parameter_prior_dict' to create proper prior dictionaries."
+                )
+
+            prior_samples_dict[index] = _sample_from_prior(
+                prior_type, prior_parameters, n_samples, rng
+            )
+
+        return prior_samples_dict
```

Comment:
Is this the most useful format to return the prior samples in? Elsewhere in pyPESTO, samples are arrays of e.g. n_parameters x n_samples (or transpose...). The dictionary means that users will need to construct the vectors themselves, right?